apps/storybook/src/Selection.stories.tsx

```diff
@@ -1,89 +1,137 @@
 import {
-  LineSelectionMesh,
   PanMesh,
-  ZoomMesh,
-  RectSelectionMesh,
+  SelectionLine,
+  SelectionMesh,
+  SelectionRect,
+  useAxisSystemContext,
   VisCanvas,
+  ZoomMesh,
 } from '@h5web/lib';
+import type {
+  ModifierKey as ModifierKeyType,
+  SelectionPoints,
+} from '@h5web/lib/src/vis/models';
 import type { Meta, Story } from '@storybook/react';
 import { format } from 'd3-format';
+import type { ReactNode } from 'react';
 import { useState } from 'react';
 import type { Vector2 } from 'three';
 
 import FillHeight from './decorators/FillHeight';
 
 interface TemplateProps {
-  selection?: 'line' | 'rectangle';
+  selectionType: 'line' | 'rectangle';
   disablePan?: boolean;
   disableZoom?: boolean;
-  modifierKey?: 'Alt' | 'Control' | 'Shift';
+  modifierKey?: ModifierKeyType;
   color?: string;
+  useWorldCoordinates?: boolean;
+  children?: (selection?: SelectionPoints) => ReactNode;
 }
 
 function vectorToStr(vec: Vector2) {
   return `(${format('.2f')(vec.x)}, ${format('.2f')(vec.y)})`;
 }
 
+interface Props {
+  selectionType: 'line' | 'rectangle';
+  startPoint: Vector2;
+  endPoint: Vector2;
+  color?: string;
+  hasWorldCoordinates?: boolean;
+}
+
+function SelectionComponent(props: Props) {
+  const { selectionType, startPoint, endPoint, color, hasWorldCoordinates } =
+    props;
+  const { dataToWorld } = useAxisSystemContext();
+
+  const Component = selectionType === 'line' ? SelectionLine : SelectionRect;
+
+  return (
+    <Component
+      color={color}
+      startPoint={hasWorldCoordinates ? startPoint : dataToWorld(startPoint)}
+      endPoint={hasWorldCoordinates ? endPoint : dataToWorld(endPoint)}
+    />
+  );
+}
+
 const Template: Story<TemplateProps> = (args) => {
   const {
-    selection,
+    selectionType,
     disablePan = true,
     disableZoom = true,
     modifierKey,
     color,
+    useWorldCoordinates,
+    children,
   } = args;
 
-  const [selectedVectors, setSelectedVectors] = useState<[Vector2, Vector2]>();
+  const [selection, setSelection] = useState<SelectionPoints>();
+  const [childrenSelection, setChildrenSelection] = useState<SelectionPoints>();
+  const resetSelection = () => {
+    setSelection(undefined);
+    setChildrenSelection(undefined);
+  };
 
   return (
-    <>
-      {selection && (
-        <p style={{ textAlign: 'center' }}>
-          {selectedVectors
-            ? `Selection from ${vectorToStr(
-                selectedVectors[0]
-              )} to ${vectorToStr(selectedVectors[1])}`
-            : 'No selection'}
-        </p>
-      )}
-      <VisCanvas
-        abscissaConfig={{ visDomain: [-10, 0], showGrid: true }}
-        ordinateConfig={{ visDomain: [50, 100], showGrid: true }}
+    <VisCanvas
+      title={
+        selection
+          ? `Selection from ${vectorToStr(
+              selection.startPoint
+            )} to ${vectorToStr(selection.endPoint)}`
+          : 'No selection'
+      }
```

Comment:
I changed the story according to https://github.com/silx-kit/h5web/pull/982/files#r812992731. It indeed feels better.